### PR TITLE
Remove collection meta after GC finished

### PR DESCRIPTION
**Pull request by longjiquan**
issue: #21416 
Signed-off-by: longjiquan <jiquan.long@zilliz.com>

### CODE REVIEW

File: internal/kv/mocks/MetaKv.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.16.0. DO NOT EDIT.
+// Code generated by mockery v2.14.0. DO NOT EDIT.
```
Comment (jaime0815) [inline internal/kv/mocks/MetaKv.go:1]: Suggest upgrading mockery to v2.16,  or the code will be formated

File: internal/metastore/mocks/DataCoordCatalog.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.16.0. DO NOT EDIT.
+// Code generated by mockery v2.14.0. DO NOT EDIT.
```
Comment (czs007) [inline internal/metastore/mocks/DataCoordCatalog.go:1]: Do not introduce modifications that are not intended

File: internal/proto/etcd_meta.proto [hunk 3]
```diff
@@ -25,13 +25,26 @@ enum CollectionState {
   CollectionCreating = 1;
   CollectionDropping = 2;
   CollectionDropped = 3;
+
+  /*
+  CollectionDroppingAndReleased = 4;
```
Comment (czs007) [inline internal/proto/etcd_meta.proto:30]: Do not introduce modifications that are not intended

**Pull request merged**